src/main/java/teammates/logic/core/FeedbackSessionsLogic.java

```diff
@@ -457,37 +461,110 @@ public FeedbackSessionQuestionsBundle getFeedbackSessionQuestionsForStudent(
             StudentAttributes student,
             Map<FeedbackQuestionAttributes, List<FeedbackResponseAttributes>> bundle,
             Map<String, Map<String, String>> recipientList,
             FeedbackQuestionAttributes question,
-            InstructorAttributes instructorGiver, StudentAttributes studentGiver)
+            Set<String> hiddenInstructorEmails)
             throws EntityDoesNotExistException {
         List<FeedbackResponseAttributes> responses =
                 frLogic.getFeedbackResponsesFromStudentOrTeamForQuestion(
                         question, student);
         Map<String, String> recipients =
-                fqLogic.getRecipientsForQuestion(question, userEmail, instructorGiver, studentGiver);
+                fqLogic.getRecipientsForQuestion(question, userEmail, null, student);
+
+        removeHiddenInstructors(question, responses, recipients, hiddenInstructorEmails);
+
         normalizeMaximumResponseEntities(question, recipients);
 
         bundle.put(question, responses);
         recipientList.put(question.getId(), recipients);
     }
-    
+
+    /**
+     * Removes instructors who are not displayed to students from
+     * {@code recipients}. Responses to the hidden instructors are also removed
+     * from {@code responses}.
+     * 
+     * @param question
+     *            the feedback question
+     * @param responses
+     *            a {@link List} of feedback responses to the {@code question}
+     * @param recipients
+     *            a {@link Map} that maps the emails of the recipients to their
+     *            names
+     * @param hiddenInstructorEmails
+     *            a {@link Set} of instructors who are not displayed to students
+     */
+    private void removeHiddenInstructors(FeedbackQuestionAttributes question,
+                                         List<FeedbackResponseAttributes> responses,
+                                         Map<String, String> recipients, 
+                                         Set<String> hiddenInstructorEmails) {
+ 
+        boolean noChangeRequired = hiddenInstructorEmails == null
+                                   || hiddenInstructorEmails.isEmpty() 
+                                   || question.getRecipientType() != FeedbackParticipantType.INSTRUCTORS;
+
+        if (noChangeRequired) {
+            return;
+        }
+
+        for (String instructorEmail : hiddenInstructorEmails) {
+
+            if (recipients.containsKey(instructorEmail)) {
+                recipients.remove(instructorEmail);
+            }
+
+            // Remove responses to the hidden instructors if they have been stored already
+            Iterator<FeedbackResponseAttributes> iterResponse = responses.iterator();
+
+            while (iterResponse.hasNext()) {
+
+                FeedbackResponseAttributes response = iterResponse.next();
+
+                if (response.recipientEmail.equals(instructorEmail)) {
+                    iterResponse.remove();
```

Comment:
Using an iterator to add/remove items is not always safe. I hope this one is safe.